presto-main/src/main/java/com/facebook/presto/sql/planner/PlanNodeCanonicalInfo.java

```diff
@@ -85,12 +100,62 @@ public static List<CanonicalPlanWithInfo> getCanonicalInfo(
                 }
                 PlanNode statsEquivalentPlanNode = node.getStatsEquivalentPlanNode().get();
                 Optional<String> hash = planCanonicalInfoProvider.hash(session, statsEquivalentPlanNode, strategy, true);
-                Optional<List<PlanStatistics>> inputTableStatistics = planCanonicalInfoProvider.getInputTableStatistics(session, statsEquivalentPlanNode, strategy, true);
-                if (hash.isPresent() && inputTableStatistics.isPresent()) {
-                    result.add(new CanonicalPlanWithInfo(new CanonicalPlan(statsEquivalentPlanNode, strategy), new PlanNodeCanonicalInfo(hash.get(), inputTableStatistics.get())));
+                Optional<List<PlanStatistics>> inputTableStatistics = planCanonicalInfoProvider.getInputTableStatistics(session, statsEquivalentPlanNode, strategy, false, true);
+                Optional<List<InputTableCacheKey>> inputTableCacheKeyList = planCanonicalInfoProvider.getInputTableCacheKey(session, statsEquivalentPlanNode, strategy, true);
+                if (hash.isPresent()) {
+                    result.add(new CanonicalPlanWithInfo(new CanonicalPlan(statsEquivalentPlanNode, strategy), new PlanNodeCanonicalInfo(hash.get(), inputTableStatistics, inputTableCacheKeyList)));
                 }
             }
         }
         return result.build();
     }
+
+    public static class InputTableCacheKey
```

Comment:
Since InputTableCacheKey is now a member of the PlanNodeCanonicalInfo class, I think it's better to move the definition here